lib/lib-table-cluster/src/test/java/org/sagebionetworks/table/cluster/SQLTranslatorUtilsTest.java

```diff
@@ -749,77 +744,63 @@ public void testReadWithHeadersWithoutEtagRow() throws SQLException{
-		assertEquals(TableConstants.JOIN_NOT_SUPPORTED_IN_THIS_CONTEX_MESSAGE, message);
-	}
-	
-	@Test
-	public void testTranslateTableReferenceWithJoin() throws ParseException{
-		FromClause element = new TableQueryParser("FROM syn123 join syn456").fromClause();
-		String message = assertThrows(IllegalArgumentException.class, ()->{
-			SQLTranslatorUtils.translate(element);
-		}).getMessage();
-		assertEquals(TableConstants.JOIN_NOT_SUPPORTED_IN_THIS_CONTEX_MESSAGE, message);
-	}
-	
-	@Test
-	public void testTranslateSelectSimple() throws ParseException{
-		ValueExpressionPrimary column = new TableQueryParser("foo").valueExpressionPrimary();
-		SQLTranslatorUtils.translateSelect(column, columnMap);
-		assertEquals("_C111_", column.toSql());
-	}
-	
-	@Test
-	public void testTranslateSelectFunction() throws ParseException{
-		ValueExpressionPrimary column = new TableQueryParser("max(foo)").valueExpressionPrimary();
-		SQLTranslatorUtils.translateSelect(column, columnMap);
-		assertEquals("MAX(_C111_)", column.toSql());
-	}
-	
-	@Test
-	public void testTranslateSelectRowId() throws ParseException{
-		ValueExpressionPrimary column = new TableQueryParser("ROW_ID").valueExpressionPrimary();
-		SQLTranslatorUtils.translateSelect(column, columnMap);
-		assertEquals("ROW_ID", column.toSql());
+		// call under test
+		Optional<TableNameCorrelation> optional = SQLTranslatorUtils.translateTableName(tableNameCorrelation, mapper);
+		assertTrue(optional.isPresent());
+		assertEquals("T123",optional.get().toSql());
 	}
-	
+
 	@Test
-	public void testTranslateSelectConstant() throws ParseException{
-		ValueExpressionPrimary column = new TableQueryParser("'constant'").valueExpressionPrimary();
-		SQLTranslatorUtils.translateSelect(column, columnMap);
-		assertEquals("'constant'", column.toSql());
+	public void testTranslateTableNameWithVerion() throws ParseException{
+		QuerySpecification model = new TableQueryParser("select * from syn123.456").querySpecification();
+		Map<IdAndVersion, List<ColumnModel>> map = new LinkedHashMap<>();
+		map.put(IdAndVersion.parse("syn123.456"), Arrays.asList(columnNameMap.get("foo"), columnNameMap.get("has space")));
+		TableAndColumnMapper mapper = new TableAndColumnMapper(model, new TestSchemaProvider(map));
+		TableNameCorrelation tableNameCorrelation = model.getFirstElementOfType(TableNameCorrelation.class);
+		
+		// call under test
+		Optional<TableNameCorrelation> optional = SQLTranslatorUtils.translateTableName(tableNameCorrelation, mapper);
+		assertTrue(optional.isPresent());
+		assertEquals("T123_456",optional.get().toSql());
 	}
 	
 	@Test
-	public void testTranslateSelectDouble() throws ParseException{
-		ValueExpressionPrimary column = new TableQueryParser("aDouble").valueExpressionPrimary();
-		SQLTranslatorUtils.translateSelect(column, columnMap);
-		assertEquals("CASE WHEN _DBL_C777_ IS NULL THEN _C777_ ELSE _DBL_C777_ END", column.toSql());
+	public void testTranslateTableNameWithJoin() throws ParseException {

```

Comment:
Let us add a test where we have joins with tables with the same name (potentially not subsequent index)